stackit/internal/services/mongodbflex/mongodbflex_acc_test.go

```diff
@@ -0,0 +1,260 @@
+					resource.TestCheckResourceAttr("stackit_mongodbflex_instance.instance", "version", instanceResource["version"]),
+					resource.TestCheckResourceAttr("stackit_mongodbflex_instance.instance", "options.type", instanceResource["options_type"]),
+				),
+			},
+			// data source
+			{
+				Config: fmt.Sprintf(`
+					%s
+
+					data "stackit_mongodbflex_instance" "instance" {
+						project_id     = stackit_mongodbflex_instance.instance.project_id
+						instance_id    = stackit_mongodbflex_instance.instance.instance_id
+					}
+					`,
+					configResources(),
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Instance data
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "project_id", instanceResource["project_id"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "name", instanceResource["name"]),
+					resource.TestCheckResourceAttrPair(
+						"data.stackit_mongodbflex_instance.instance", "project_id",
+						"stackit_mongodbflex_instance.instance", "project_id",
+					),
+					resource.TestCheckResourceAttrPair(
+						"data.stackit_mongodbflex_instance.instance", "instance_id",
+						"stackit_mongodbflex_instance.instance", "instance_id",
+					),
+
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "acl.#", "1"),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "acl.0", instanceResource["acl"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "flavor.id", instanceResource["flavor_id"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "flavor.description", instanceResource["flavor_description"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "flavor.cpu", instanceResource["flavor_cpu"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "flavor.ram", instanceResource["flavor_ram"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "replicas", instanceResource["replicas"]),
+					resource.TestCheckResourceAttr("data.stackit_mongodbflex_instance.instance", "options.type", instanceResource["options_type"]),
+				),
+			},
+			// Import
+			{
+				ResourceName: "stackit_mongodbflex_instance.instance",
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					r, ok := s.RootModule().Resources["stackit_mongodbflex_instance.instance"]
+					if !ok {
+						return "", fmt.Errorf("couldn't find resource stackit_mongodbflex_instance.instance")
+					}
+					instanceId, ok := r.Primary.Attributes["instance_id"]
+					if !ok {
+						return "", fmt.Errorf("couldn't find attribute instance_id")
+					}
+
+					return fmt.Sprintf("%s,%s", testutil.ProjectId, instanceId), nil
+				},
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update
+			{
+				Config: fmt.Sprintf(`
```

Comment:
Pass whatever fields are updated (I think it's just `version`) as input to `configResources`, and use it here instead of rewriting the whole config